amethyst_utils/src/removal.rs

```diff
@@ -0,0 +1,59 @@
+use amethyst_assets::{PrefabData, PrefabError};
+use amethyst_core::specs::{Component, DenseVecStorage, Entity, Join, ReadStorage, WriteStorage};
+use amethyst_core::specs::world::EntitiesRes;
+
+use std::result::Result;
+
+/// A marker `Component` used to remove entities and clean up your scene.
+/// The generic parameter `I` is the type of id you want to use.
+/// Generally a int or an enum.
+pub struct Removal<I> {
+    id: I,
```

Comment:
Why not directly use the type as a marker?

Reply:
I guess it's fine right now because it will be better for scripts, so I'd keep it that way. But any reason?

Reply:
Its made to be used like this:
entity1: Removal<Int>(0)
entity2: Removal<Int>(0)
entity3: Removal<Int>(1)

exec_removal(entities, storage, 0)

Its where the whole grouping idea I talked about started from.